cpp/src/barretenberg/proof_system/circuit_builder/goblin_translator_circuit_builder.test.cpp

```diff
@@ -0,0 +1,430 @@
+        v_squared_witnesses[0] * p_y_witnesses[0] + v_cubed_witnesses[0] * z_1_lo + v_quarted_witnesses[0] * z_2_lo +
+        quotient_witnesses[0] * neg_modulus_limbs[0] - remainder_witnesses[0]; // This covers the lowest limb
+    info("LW1:", low_wide_relation_limb_part_1);
+    Fr low_wide_relation_limb =
+        low_wide_relation_limb_part_1 +
+        (previous_accumulator_witnesses[1] * x_witnesses[0] + previous_accumulator_witnesses[0] * x_witnesses[1] +
+         v_witnesses[1] * p_x_witnesses[0] + p_x_witnesses[1] * v_witnesses[0] +
+         v_squared_witnesses[1] * p_y_witnesses[0] + v_squared_witnesses[0] * p_y_witnesses[1] +
+         v_cubed_witnesses[1] * z_1_lo + z_1_hi * v_cubed_witnesses[0] + v_quarted_witnesses[1] * z_2_lo +
+         v_quarted_witnesses[0] * z_2_hi + quotient_witnesses[0] * neg_modulus_limbs[1] +
+         quotient_witnesses[1] * neg_modulus_limbs[0] - remainder_witnesses[1]) *
+            shift_1; // And this covers the limb shifted by 68
+    for (auto& limb : quotient_witnesses) {
+        info("Q: ", limb);
+    }
+    // Treating accumulator as 254-bit constrained value
+    constexpr auto max_limb_size = (uint512_t(1) << NUM_LIMB_BITS) - 1;
+    constexpr auto shift_1_u512 = uint512_t(shift_1);
+    constexpr auto op_max_size = uint512_t(4);
+    constexpr uint512_t lwl_maximum_value = op_max_size + (max_limb_size * max_limb_size) * ((shift_1_u512 * 12) + 6);
+    constexpr uint512_t lwl_maximum_value_constraint =
+        (lwl_maximum_value >> (2 * NUM_LIMB_BITS)).lo +
+        uint256_t(uint64_t((lwl_maximum_value % uint512_t(1) << (2 * NUM_LIMB_BITS)) != 0));
+    constexpr auto lwl_range_consraint_size = lwl_maximum_value_constraint.get_msb() + 1;
+    info("Low limb range constraint: ", lwl_range_consraint_size);
+    // Low bits have to be zero
+    ASSERT(uint256_t(low_wide_relation_limb).slice(0, 2 * NUM_LIMB_BITS) == 0);
+
+    Fr low_wide_relation_limb_divided = low_wide_relation_limb * shift_2_inverse;
+    // We need to range constrain the low_wide_relation_limb_divided
+    constexpr size_t NUM_LAST_BN254_LIMB_BITS = modulus_u512.get_msb() + 1 - NUM_LIMB_BITS * 3;
+
+    constexpr auto max_high_limb_size = (uint512_t(1) << NUM_LAST_BN254_LIMB_BITS) - 1;
+    constexpr uint512_t hwl_maximum_value =
+        lwl_maximum_value_constraint + (max_limb_size * max_limb_size) * 16 +
+        (max_limb_size * max_limb_size * 10 + max_limb_size * max_high_limb_size * 10) * shift_1_u512;
+    constexpr uint512_t hwl_maximum_value_constraint =
+        (hwl_maximum_value >> (2 * NUM_LIMB_BITS)).lo +
+        uint256_t(uint64_t((hwl_maximum_value % uint512_t(1) << (2 * NUM_LIMB_BITS)) != 0));
+    constexpr auto hwl_range_consraint_size = hwl_maximum_value_constraint.get_msb() + 1;
+    info(hwl_range_consraint_size);
+    // 4 high combinations = 8 ml*ml + 8 ml*last_ml. 2 low combinations = 2*ml*ml + 2*ml*last_ml
+    Fr high_wide_relation_limb =
```

Comment:
This is another chuck that's quite hard to parse. Worth adding comments and/or breaking into smaller components for readability but that can wait as I know this is a WiP

Reply:
This is the same thing as in the check_circuit. Will add comments later